### PR TITLE
Ghost Cafe Remodel

### PR DESCRIPTION
**Pull request by Thlumyn**
## About The Pull Request

Collaboration with @TheSS13Melon 

Changes up the ghost cafe, moving and condensing the existing cabins while adding some new areas, including a mini-station, an ashwalker camp and an ice cat camp. 

It also removes some unused cafe areas (on the code end). 

## How This Contributes To The Skyrat Roleplay Experience

A lot of people have scenes that make sense to happen on a space station - either continuations of scenes that happened in previous rounds, or new scenes altogether. While going to the cafe and pretending that you were still in the bridge, or still in the CE office, or still in the interrogation room, was well and fine, adding simplified mini versions of those might be nice for people to stay immersed. Some of these were already sort of there, such as a surgery room, janitor room and cell that seemed to be more or less placed randomly into the cafe area. The cafe is still supposed to be a cafe however, and we tried to balance the outdoor cafe-y theme with the station parts.

This also adds areas such as a sauna/gym that people can use in the cafe. And a camp for the ice cats and ashwalkers. 

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary>
  
image

image

image

image

image

image

image

image

image

</details>

## Changelog

:cl:
add: Added and moved some areas in the ghost cafe.
add: Added an icewalker section to the ghost cafe.
add: Added an ashwalker section to the ghost cafe.
/:cl:

### CODE REVIEW

**Comment by burgerenergy** [comment]
Not to say that this is a bad idea or anything, but I kinda think using the infini-dorms/hotel/whatever so people can just create the zones might be easier. That said the foot print also doesn't matter, so eh.

**Comment by sqnztb** [comment]
Okay. A few... issues.

image
The ID console is a full console, and cafe visitors spawn with with access. Let's not have that.

image
Engineering radio headset in the locker. Also bad.

image
Hidden beach cabin is dark. 

image
Another ID console. Power monitoring console which does nothing. I'll also point out at this point that ALL OF THESE consoles (including the civilian ones scattered around) all have NT Software Hub and could download things like the **science hub**, chat client, bounty board requests, etc. More cafe-station interaction points that we don't want.

image
I know this is an issue on the live server (which I'll probably fix in a few minutes), but paintings need to be moved off of the wall tiles and pixel-shifted properly (so they don't show through the walls).

image
Weird transition here.

image
This looks awkward since you moved the wall up but didn't shift the shower up.

Vox box still isn't safe for vox.

Personal gripe, but stepping out of the shuttle and into a waiting area that looks like medbay feels awkward.

The ashwalker camp is cool. The icecat camp is nice, though I don't approve of the pile of surgery tools laying on the rack. I appreciate a lot of the effort that went into doing this, but I don't really think the mini-station is needed, at least in the current form, and some of the moves of some of the existing rooms make for some really awkward spots like - 
image

**Comment by Thlumyn** [comment]
Thanks for looking at it @sqnztb and thanks for the quick response!

I've made changes to those things you have pointed out. 

I added fake consoles based on the fakeid and fakesec consoles from the showcase. I completely forgot it was possible to add more programs onto civilian consoles, o o p s. 

I changed the voxbox to nitrogen atmos. 

As far as that entryway - it was supposed to be "clean white aesthetic" with blue trim for NT, but I see totally how it looks like medbay now that you point it out, I darkened the floor tiles to match the NT area to the left. 

As far as the existence of the ministation and the placement of the old cabins go, that's for @TheSS13Melon  to respond.

**Comment by Thlumyn** [comment]
Oops I missed a computer

File: modular_skyrat/modules/ghostcafe/code/ghost_role_spawners.dm [hunk 1]
```diff
@@ -28,8 +28,8 @@
 		var/area/A = get_area(src)
 		//new_spawn.AddElement(/datum/element/ghost_role_eligibility, free_ghosting = TRUE) SKYRAT PORT -- Needs to be completely rewritten
 		new_spawn.AddElement(/datum/element/dusts_on_catatonia)
-		new_spawn.AddElement(/datum/element/dusts_on_leaving_area,list(A.type, /area/misc/hilbertshotel, /area/centcom/holding/cafe, /area/centcom/holding/cafewar, /area/centcom/holding/cafebotany,
-		/area/centcom/holding/cafebuild, /area/centcom/holding/cafevox, /area/centcom/holding/cafedorms, /area/centcom/holding/cafepark, /area/centcom/holding/cafeplumbing))
+		new_spawn.AddElement(/datum/element/dusts_on_leaving_area,list(A.type, /area/misc/hilbertshotel, /area/centcom/holding/cafe,
```
Comment (YakumoChen) [inline modular_skyrat/modules/ghostcafe/code/ghost_role_spawners.dm:31]: Hilberts Hotel is a valid location reachable by normal players through a space ruin, I do not recommend including it here

Comment (Thlumyn) [reply modular_skyrat/modules/ghostcafe/code/ghost_role_spawners.dm:31]: That was already there, I didn't mess with that.
The ghost cafe already has access to hilbert's hotel, and a ton of people use the hotel rooms there for scenes, so we wouldn't be removing that.

**Comment by Thlumyn** [comment]
Well, if Deadmon doesn't respond to that soon, I might just go in and make some changes myself.

**Comment by sqnztb** [comment]
Much, much, much better looking.

Okay. Couple of tiny requests.

image
Give a button for curtain toggling. Just makes it a bit of a hassle to close them all otherwise.

image
It's probably okay to leave this as an actual surgery computer? We've got one already in the cafe. Maybe find a place to shove a NanoMed, NanoDrug Plus, and a few medkits like we have currently. I see no harm in leaving them as a *just in case* situation.

Everything else looks good to go as far as I'm concerned.

**Comment by Thlumyn** [comment]
Okay @sqnztb and @TheSS13Melon  I have made some changes to make it a bit more cafe-like 

I condensed medbay and robotics together into a "Healthcare Center".
image

I removed the bridge and condensed security down into one room and one cell named "Cafe Security" and also added a little river in the now empty area.
image

I removed the engineering section and replaced it with a spa that includes a "hot tub", barber area and massage area. I don't know, if I were on vacation in a cafe resort that's what I'd want to see. I also renamed the generic command office into "Cafe Manager Office". I suppose that way its more cafe immersive but people can still use it for those head of staff office RP if they want. 
image

I moved around some of the cabins that had been around just slightly - mostly this one, and added a little patio. To try and make that hallway less clunky. 
image

Also fixed some of the paintings while I was at it, replaced the photocopier with a free one, and added a towel bin near the pool.

**Comment by Thlumyn** [comment]
Oh oops, I didn't see your response, I accidentally left the comment open but unsent.

**Comment by Thlumyn** [comment]
Okay, I added the curtain button and those suggestions (the surgery computer, medkits and medical vendors)
image

**Comment by theselfish** [comment]
image
You seem to have left a bench in the room near botany.
image
If this is meant to be a grilling pit, you should make it the grille version.

**Comment by SomeRandomOwl** [comment]
dreamseeker_GLAhjg2i3X

dreamseeker_UZXZW4WPLy

The news caster should be removed and that civilian console should be replaced with a fake. as they both enable station interaction

**Comment by Thlumyn** [comment]
Thanks, I didn't even see those.

**Review by Gandalf2k15 — APPROVED** [review]
Looks great!

**Pull request merged**